kubectl-ktop: init at 0.3.5

ivankovnatsky opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ivankovnatsky committed:
kubectl-ktop: init at 0.3.5

IndeedNotJames commented on pkgs/applications/networking/cluster/kubectl-ktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+
+  vendorSha256 = "sha256-IbWdq6t9tgly7MRRIRJK5sDHus+sF0AT843aI4quO3c=";
+
+  postConfigure = ''
+    rm -rf ./.ci ./hack
+  '';
```
I guess you added this because it otherwise tries to build those sub packages?
Use the following instead:

```suggestion
  subPackages = [ "." ];
```

IndeedNotJames commented on pkgs/applications/networking/cluster/kubectl-ktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+
+buildGoModule rec {
+  pname = "ktop";
+  version = "0.3.0";
```
```suggestion
  version = "0.3.5";
```

https://github.com/vladimirvivien/ktop/releases/tag/v0.3.5 is out now (don't forget to update both hashes)

IndeedNotJames commented on pkgs/applications/networking/cluster/kubectl-ktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "ktop";
```
```suggestion
  pname = "kubectl-ktop";
```

IndeedNotJames commented on pkgs/applications/networking/cluster/kubectl-ktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+  meta = with lib; {
+    description = "A top-like tool for your Kubernetes clusters";
+    homepage = "https://github.com/vladimirvivien/ktop";
+    changelog = "https://github.com/vladimirvivien/ktop/releases/tag/${version}";
```
```suggestion
    changelog = "https://github.com/vladimirvivien/ktop/releases/tag/v${version}";
```

IndeedNotJames commented on pkgs/applications/networking/cluster/kubectl-ktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+    rm -rf ./.ci ./hack
+  '';
+
+  doCheck = false;
```
Even if the upstream project has no tests at the moment, you can still just leave this on.
It won't error out just because of this.

And because `true` is the default for this flag here, you can just remove it all together:
```suggestion
```

IndeedNotJames commented:
Please squash your two commits into a single titled `kubectl-ktop: init at 0.3.5`.
Thank you! :)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2108

SuperSandro2000 commented on pkgs/applications/networking/cluster/kubectl-ktop/default.nix:
```diff
@@ -0,0 +1,36 @@
+    "-X github.com/vladimirvivien/ktop/buildinfo.GitSHA=${src.rev}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/ktop $out/bin/kubectl-ktop
+  '';
+
```
```suggestion
```
the upstream docs mention ktop as a binary name

IndeedNotJames replied:
`kubectl <subcommand>` auto-discovery checks for binaries named `kubectl-<subcommand>` in the `$PATH`.
E.g. `kubectl ktop` => `kubectl-ktop`.

`nix run`/`lib.getExe` correctly picks `kubectl-ktop` because of the `pname` as well.

So what do you suggest?
A symlink?

ivankovnatsky replied:
We already do similar manipulations for other kubectl plugins.

Even though ktop works normally as a single binary, I would probably keep the kubectl-<plugin> naming scheme as @IndeedNotJames suggests.

SuperSandro2000 replied:
Then we should create a symlink because it is confusing if upstream docs no longer apply.

ivankovnatsky commented on pkgs/applications/networking/cluster/kubectl-ktop/default.nix:
```diff
@@ -0,0 +1,36 @@
+  ];
+
+  postInstall = ''
+    mv $out/bin/ktop $out/bin/kubectl-ktop
```
```suggestion
    ln -s $out/bin/ktop $out/bin/kubectl-ktop
```

IndeedNotJames commented:
Please squash your commits again :)
Ref https://github.com/NixOS/nixpkgs/pull/226032#issuecomment-1512488483

ivankovnatsky commented:
> Please squash your commits again :) Ref [#226032 (comment)](https://github.com/NixOS/nixpkgs/pull/226032#issuecomment-1512488483)

Sorry, I do this for more clear review process. I don't have permissions to view how people usually merge, but I guess they can easily merge with squash.

Anyway, squashed and updated. Thanks.

IndeedNotJames commented:
In nixpkgs, PRs get merged, not squashed or rebased into master (or any different branch).

This is done in order to keep complete commit messages and GPG signatures intact.
See https://github.com/NixOS/nixpkgs/blob/372e0f8efc5ee4d73ab7197c0312e7ee9d13d997/CONTRIBUTING.md#writing-good-commit-messages

It has its own downsides, for sure, and it's also kinda uncommon for other projects on GitHub :woman_shrugging: 

I have no bad intentions, it's just the way it is :^)

---

Anyhow...
thanks for working on this!
I think this turned out great!

And with the addition of the symlink @SuperSandro2000 will probably be happy as well :)

kira-bruneau commented:
Result of `nixpkgs-review pr 226032` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubectl-ktop</li>
  </ul>
</details>

kira-bruneau reviewed (approved):
Thanks @ivankovnatsky!!
